vectordb_orm/base.py

```diff
@@ -0,0 +1,178 @@
+        # class to return the value of the field (or default to AttributeCompare via __getattr__ in the case of
+        # a direct class attribute), not the field configuration itself.
+        cls._type_configuration = {}
+        for key in cls.__annotations__.keys():
+            attribute_value = getattr(cls, key)
+            cls._type_configuration[key] = attribute_value if isinstance(attribute_value, BaseField) else None
+            delattr(cls, key)
+
+    @classmethod
+    def collection_name(self) -> str:
+        if not hasattr(self, '__collection_name__'):
+            raise ValueError(f"Class {self.__name__} does not have a collection name, specify `__collection_name__` on the class definition.")
+        return self.__collection_name__
+
+    @classmethod
+    def embedding_dim(cls) -> int:
+        return cls.__embedding_dim__
+
+    @classmethod
+    def _create_collection(cls, milvus_client: Milvus):
+        cls._assert_has_primary()
+
+        fields: list[FieldSchema] = []
+
+        for attribute_name, type_hint in cls.__annotations__.items():
+            fields.append(
+                cls._field_schema_from_typehint(
+                    attribute_name,
+                    type_hint,
+                    cls._type_configuration.get(attribute_name)
+                )
+            )
+
+        schema = CollectionSchema(fields=fields, description=f"{cls.__name__} vectordb-generated collection")
+        collection = Collection(cls.collection_name(), schema)
+
+        # For all embeddings, create an associated index
+        for attribute_name, field_configuration in cls._type_configuration.items():
+            if isinstance(field_configuration, EmbeddingField):
+                index = {
+                    "index_type": field_configuration.index.index_type,
+                    "metric_type": field_configuration.metric_type.value,
+                    "params": field_configuration.index.get_parameters(),
+                }
+                collection.create_index("embedding", index)
+
+        return collection
+
+    @classmethod
+    def _field_schema_from_typehint(cls, name: str, type_hint: Any, field_customization: BaseField | None = None):
+        if issubclass(type_hint, np.ndarray):
+            if not isinstance(field_customization, EmbeddingField):
+                raise ValueError("Embedding typehints should be configured with vectordb.EmbeddingField")
+            return FieldSchema(name=name, dtype=DataType.FLOAT_VECTOR, dim=field_customization.dim)
+        elif issubclass(type_hint, str):
+            if not isinstance(field_customization, VarCharField):
+                raise ValueError("String typehints should be configured with vectordb.VarCharField")
+            return FieldSchema(name=name, dtype=DataType.VARCHAR, max_length=field_customization.max_length)
+        elif issubclass(type_hint, int):
+            # We currently only support ints as the primary key
```

Comment:
suggest enforcing this in the `PrimaryKeyField` class

Reply:
actually i think i interpreted this backwards- the primary key field can be non-int but the only field that can be int is the primary key field? did i get that right?